frontend/src/store/items.ts

```diff
@@ -81,35 +81,19 @@ class ItemsStore {
    * @param payload
    * @returns - The reactive references
    */
-  public add = (
-    payload: BaseItemDto | BaseItemDto[]
-  ): BaseItemDto | BaseItemDto[] => {
-    const isArray = Array.isArray(payload);
-    const itemArray = isArray ? payload : [payload];
-
-    const res: BaseItemDto[] = [];
-
-    for (const item of itemArray) {
-      if (!item.Id) {
-        throw new Error("One item doesn't have an id");
-      }
-
-      state.value.byId[item.Id] = item;
-
-      const fetched = this.getItemById(item.Id);
-
-      if (!fetched) {
-        throw new Error('Expected to receive newly added item to store');
-      }
-
-      res.push(fetched);
+  public add = (item: BaseItemDto): BaseItemDto => {
+    // Without an id this cannot be cached so return a non-reactive version
```

Comment:
This change is necessary since calls like getting devices or apikeys also end up coming through this function since, see my comment in `axios/index`, but they won't have a valid Id, and this was previously failing. Now if there no id the non-reactive value is still returned without being cached so those calls will continue working.

Reply:
@aweebs I like your fallback when the item doesn't have an id.

About the axios interceptor, ideally we should strictly check for everything before calling this method in the interceptor. Previously in the item store we had all the API methods (some, not all of them), which was a maintenance burden. The interceptor-way of working was a middle-ground I came up with to ease the maintenance burden, however it didn't end up working because the Vue 2 limitations. 

Just saying all of this if you want to take another challenge :): Investigate all the endpoints that returns BaseItemDto or BaseItemPerson and filter  them properly.

Reply:
I'll add it to my list of things to work on :)
I also added a fairly descriptive TODO to the interceptor so that anyone else who comes across it is also aware of the limitations and that there is work to be done there.